From 047942ebc19a015ee1fa222b9ab7725b63b4ff75 Mon Sep 17 00:00:00 2001
From: Lukasz Maniak <lukasz.maniak@intel.com>
Date: Mon, 22 Jun 2015 14:11:24 -0700
Subject: [PATCH 1/8] Fix BadDrawable error caused by DRI_glXUseXFont

Fix for throwing BadDrawable (invalid Pixmap or Window parameter)
by Xserver when calling XCreatePixmap or fill_bitmap from DRI_glXUseXFont.

currentDrawable which came from __glXGetCurrentContext was not correct in
this case, we use DefaultRootWindow(dpy) instead.

Signed-off-by: Matt Roper <matthew.d.roper@intel.com>
---
 src/glx/xfont.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/glx/xfont.c b/src/glx/xfont.c
index 00498bc..36f42a0 100644
--- a/src/glx/xfont.c
+++ b/src/glx/xfont.c
@@ -235,7 +235,7 @@ DRI_glXUseXFont(struct glx_context *CC, Font font, int first, int count, int lis
    int i;
 
    dpy = CC->currentDpy;
-   win = CC->currentDrawable;
+   win = DefaultRootWindow(dpy);
 
 #if !defined(GLX_USE_APPLEGL)
    glxdraw = GetGLXDRIDrawable(CC->currentDpy, CC->currentDrawable);
-- 
2.7.4

